### PR TITLE
[Clang][RISCV] Add vundefine intrinsics for tuple types

### PR DESCRIPTION
**Pull request by eopXD**
riscv-non-isa/rvv-intrinsic-doc#288

### CODE REVIEW

File: clang/test/CodeGen/RISCV/rvv-intrinsics-autogenerated/non-policy/non-overloaded/vundefined.c [hunk 1]
```diff
@@ -7,534 +7,2189 @@
...
-// CHECK-RV64-SAME: () #[[ATTR0:[0-9]+]] {
+// CHECK-RV64-LABEL: define dso_local <vscale x 1 x half> @test_vundefined_f16mf4(
+// CHECK-RV64-SAME: ) #[[ATTR0:[0-9]+]] {
 // CHECK-RV64-NEXT:  entry:
 // CHECK-RV64-NEXT:    ret <vscale x 1 x half> poison
 //
-vfloat16mf4_t test_vundefined_f16mf4() {
-  return __riscv_vundefined_f16mf4();
-}
+vfloat16mf4_t test_vundefined_f16mf4() { return __riscv_vundefined_f16mf4(); }
```
Comment (kito-cheng) [inline clang/test/CodeGen/RISCV/rvv-intrinsics-autogenerated/non-policy/non-overloaded/vundefined.c:15]: Why this format changed here? it seems unrelated to this patch?

Comment (eopXD) [reply clang/test/CodeGen/RISCV/rvv-intrinsics-autogenerated/non-policy/non-overloaded/vundefined.c:15]: I added script under rvv-intrinsic-doc to apply clang-format to auto-generated test cases.

File: clang/test/CodeGen/RISCV/rvv-intrinsics-autogenerated/non-policy/non-overloaded/vundefined.c [hunk 2]
```diff
@@ -1,4 +1,4 @@
-// NOTE: Assertions have been autogenerated by utils/update_cc_test_checks.py UTC_ARGS: --version 2
+// NOTE: Assertions have been autogenerated by utils/update_cc_test_checks.py UTC_ARGS: --version 3
```
Comment (kito-cheng) [inline clang/test/CodeGen/RISCV/rvv-intrinsics-autogenerated/non-policy/non-overloaded/vundefined.c:1]: Does this cause those changes?
```diff
-// CHECK-RV64-LABEL: define dso_local <vscale x 1 x half> @test_vundefined_f16mf4
-// CHECK-RV64-SAME: () #[[ATTR0:[0-9]+]] {
+// CHECK-RV64-LABEL: define dso_local <vscale x 1 x half> @test_vundefined_f16mf4(
+// CHECK-RV64-SAME: ) #[[ATTR0:[0-9]+]] {
```

Comment (eopXD) [reply clang/test/CodeGen/RISCV/rvv-intrinsics-autogenerated/non-policy/non-overloaded/vundefined.c:1]: I guess this is something new inside the `update_cc_test_check.py` script.

**Review by kito-cheng — APPROVED** [review]
LGTM
